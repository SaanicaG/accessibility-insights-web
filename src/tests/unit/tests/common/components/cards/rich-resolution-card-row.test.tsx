// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import {
    RichResolutionCardRow,
    RichResolutionCardRowProps,
} from 'common/components/cards/rich-resolution-card-row';
import { RichResolutionContent } from 'common/components/cards/rich-resolution-content';
import * as React from 'react';
import { mockReactComponents } from 'tests/unit/mock-helpers/mock-module-helpers';
jest.mock('common/components/cards/rich-resolution-content');
describe('RichResolutionCardRow', () => {
    mockReactComponents([RichResolutionContent]);
    it.each(['check', 'fix'] as const)('renders with labelType=%s', labelType => {
        const props: RichResolutionCardRowProps = {
            deps: null,
            index: 123,
            propertyData: {
                labelType,
                contentId: 'content-id',
                contentVariables: { 'content-var-key': 'content-var-value' },
            },
        };

        const renderResult = render(<RichResolutionCardRow {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
