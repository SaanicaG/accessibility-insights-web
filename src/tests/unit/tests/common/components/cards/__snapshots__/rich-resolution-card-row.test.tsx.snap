// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichResolutionCardRow renders with labelType=check 1`] = `
<DocumentFragment>
  How to check
  <mock-richresolutioncontent
    contentid="content-id"
    contentvariables="[object Object]"
    labeltype="check"
  />
</DocumentFragment>
`;

exports[`RichResolutionCardRow renders with labelType=fix 1`] = `
<DocumentFragment>
  How to fix
  <mock-richresolutioncontent
    contentid="content-id"
    contentvariables="[object Object]"
    labeltype="fix"
  />
</DocumentFragment>
`;
